### PR TITLE
fix(security): sanitize against injection

### PR DESCRIPTION
**Pull request by sc979**
- Sanitizing http parameters to avoid injections
- Cleaning code or style

For :
20.04
19.10
19.04
18.10

### CODE REVIEW

File: service-monitoring/index.php [hunk 1]
```diff
@@ -44,16 +44,15 @@
...
     $db = $dependencyInjector['configuration_db'];
     $widgetObj = new CentreonWidget($centreon, $db);
     $preferences = $widgetObj->getWidgetPreferences($widgetId);
-    $autoRefresh = 0;
-    if (isset($preferences['refresh_interval'])) {
-        $autoRefresh = $preferences['refresh_interval'];
-    }
+    $autoRefresh = (isset($preferences['refresh_interval']) && (int)$preferences['refresh_interval'] > 0)
+        ? (int)$preferences['refresh_interval']
+        : 30;
```
Comment (callapa) [inline service-monitoring/index.php]: Use constant instead

Comment (sc979) [reply service-monitoring/index.php]: Indeed, I should have use it. Will do on next PR

File: service-monitoring/src/export.php [hunk 2]
```diff
@@ -385,7 +385,9 @@
     );
 }
 
-$autoRefresh = $preferences['refresh_interval'];
+$autoRefresh = (isset($preferences['refresh_interval']) && (int)$preferences['refresh_interval'] > 0)
+    ? (int)$preferences['refresh_interval']
+    : 30;
```
Comment (callapa) [inline service-monitoring/src/export.php]: use constant instead

File: service-monitoring/src/index.php [hunk 3]
```diff
@@ -382,7 +382,8 @@
     $query .= "ORDER BY " . $orderBy;
 }
 
-$query .= " LIMIT " . ($page * $preferences['entries']) . "," . $preferences['entries'];
+$num = filter_var($preferences['entries'], FILTER_VALIDATE_INT) ?: 10;
```
Comment (callapa) [inline service-monitoring/src/index.php]: use constant instead

File: service-monitoring/src/index.php [hunk 4]
```diff
@@ -395,8 +396,8 @@
 
 $nbRows = $dbb->numberRows();
 $data = [];
-$outputLength = $preferences['output_length'] ? $preferences['output_length'] : 50;
-$commentLength = $preferences['comment_length'] ? $preferences['comment_length'] : 50;
+$outputLength = $preferences['output_length'] ?: 50;
```
Comment (callapa) [inline service-monitoring/src/index.php]: use constant instead

File: service-monitoring/src/index.php [hunk 5]
```diff
@@ -395,8 +396,8 @@
...
+$commentLength = $preferences['comment_length'] ?: 50;
```
Comment (callapa) [inline service-monitoring/src/index.php]: use constant instead

File: service-monitoring/src/index.php [hunk 6]
```diff
@@ -579,7 +580,9 @@
     );
 }
 
-$autoRefresh = $preferences['refresh_interval'];
+$autoRefresh = (isset($preferences['refresh_interval']) && (int)$preferences['refresh_interval'] > 0)
+    ? (int)$preferences['refresh_interval']
+    : 30;
```
Comment (callapa) [inline service-monitoring/src/index.php]: use constant instead

File: service-monitoring/src/sendCmd.php [hunk 7]
```diff
@@ -134,13 +135,13 @@
...
-            $svcId = $tmp[1];
-            if ($hostId != 0 && $svcId != 0) {
+            $hostId = filter_var($tmp[0], FILTER_VALIDATE_INT) ?: 0;
+            $svcId = filter_var($tmp[1], FILTER_VALIDATE_INT) ?: 0;
+            if ($hostId !== 0 && $svcId !== 0) {
                 $hostname = $hostObj->getHostName($hostId);
                 $svcDesc = $svcObj->getServiceDesc($svcId);
                 $pollerId = $hostObj->getHostPollerId($hostId);
-                if ($cmd == 70 || $cmd == 74) {
+                if ($cmd === 70 || $cmd === 74) {
```
Comment (callapa) [inline service-monitoring/src/sendCmd.php]: 70 ? 74 ?
use constant instead

**Pull request merged**